server/src/main/java/org/elasticsearch/index/fielddata/plain/SortedNumericIndexFieldData.java

```diff
@@ -43,17 +43,28 @@ public class SortedNumericIndexFieldData extends IndexNumericFieldData {
     public static class Builder implements IndexFieldData.Builder {
         private final String name;
         private final NumericType numericType;
+        private final ValuesSourceType valuesSourceType;
         protected final ToScriptFieldFactory<SortedNumericDocValues> toScriptFieldFactory;
 
         public Builder(String name, NumericType numericType, ToScriptFieldFactory<SortedNumericDocValues> toScriptFieldFactory) {
+            this(name, numericType, numericType.getValuesSourceType(), toScriptFieldFactory);
```

Comment:
Same question here.

Reply:
The old constructor here is used in a dozen places which suggests to me that it's still useful?

Reply:
I guess. I just don't want to end up with a zillion ctors.